File: 509_fibonacci_number/Solution.java
class Solution {
    public int fib(int N) {
        if (N < 2) return N;

        int a = 0, b = 1;
        while (N-- > 1) {
            int tmp = a;
            a = b;
            b += tmp;
        }

        return b;
    }

    public static void main(String[] args) {
        System.out.println(new Solution().fib(7));
    }
}

File: 509_fibonacci_number/Solution.go
package main

import "fmt"

func fib(n int) int {
    if n < 2 { return n }

    a, b := 0, 1
    for n > 1 {
        a, b = b, a + b
        n--
    }

    return b
}

func main() {
    fmt.Println(fib(7))
}
